projects/feature-activation/0004-phased-testing.md

```diff
@@ -0,0 +1,206 @@
+2. It transitions to `STARTED` on block `N`, for reaching its `start_height`.
+3. After 2 weeks, it continues on `STARTED` on block `N + 40320`, as its `timeout_height` has not been reached yet.
+4. After 2 weeks, it gets to `FAILED` on block  `N + 2 * 40320`, as its `timeout_height` is reached and `lock_in_on_timeout` is `false`.
+
+Therefore, `NOP_FEATURE_3`'s process takes 4 weeks.
+
+
+### Validation via the Hathor Explorer
+
+By manually inspecting blocks via the Hathor Explorer interfaces described in the [Feature Activation Explorer UIs RFC](https://github.com/HathorNetwork/rfcs/blob/master/projects/feature-activation/0003-explorer-uis.md), we'll be able to validate the expected state for each block-feature pair according to the lists described above.
+
+### Validation via logging
+
+Complementing the validation via Explorer, we'll also have validation via full node logs, which will be useful to perform more complex queries to understand state transitions, specially if debugging is necessary. By using structured logging of features states for each block, we'll be able to leverage AWS CloudWatch queries to perform analysis and even export metrics.
+
+After a new block is received and processed by the full node, we'll log its height, together with its state for each NOP feature. At the end of the `HathorManager.tx_fully_validated()` method, we'll call a new `HathorManager.log_feature_states()` method, defined as follows:
+
+```python
+def log_feature_states(self, vertex: BaseTransaction) -> None:
+    if not isinstance(vertex, Block):
+        return
+
+    feature_descriptions = self.feature_service.get_bits_description(block=vertex)
+    state_by_feature = {
+        feature.value: description.state.value
+        for feature, description in feature_descriptions.items()
+    }
+
+    self.log.info(
+        'New block accepted with feature activation states',
+        block_height=vertex.get_height(),
+        features_states=state_by_feature
+    )
+```
+
+This logging will be removed after Phase 1 is complete.
+
+## Phase 2
+
+Phase 2 can start as soon as Phase 1 ends. Let `M` be a block height for the beginning of Phase 2. It'll define a new `INCREASE_MAX_MERKLE_PATH_LENGTH` feature that will actually affect how the full node verifies merged mining blocks.
+
+Currently, there's a `MAX_MERKLE_PATH_LENGTH: int = 12` constant in `aux_pow.py`. This constant will be replaced by a function call to `get_max_merkle_path_length()`, with its reference implementation as follows:
+
+```python
+def get_max_merkle_path_length(feature_service: FeatureService, block: Block) -> int:
+    if feature_service.is_feature_active(
+      block=block,
+      feature=Feature.INCREASE_MAX_MERKLE_PATH_LENGTH
+    ):
+        return NEW_MAX_MERKLE_PATH_LENGTH
+
+    return OLD_MAX_MERKLE_PATH_LENGTH
+```
+
+Where this function is actually defined will be left as an implementation detail. TODO: what's a good value for `NEW_MAX_MERKLE_PATH_LENGTH`?
```

Comment:
@jansegre do we know what value we want to use for `NEW_MAX_MERKLE_PATH_LENGTH`?